File: 007_reverse_integer/Solution.go
package main

import (
	"fmt"
	"math"
)

func reverse(x int) int {
	result := 0.0
	rightLimit := math.Pow(2, 31) - 1
	leftLimit := -rightLimit - 1

	for x != 0 {
		result = result*10.0 + float64(x%10.0)
		if result > rightLimit || result < leftLimit {
			return 0
		}
		x /= 10.0
	}

	return int(result)
}

func main() {
	fmt.Println(reverse(-123))
}

File: 007_reverse_integer/Solution.py
class Solution:
    def reverse(self, x: int) -> int:
        limit = 2 ** 31 - 1 if x >= 0 else 2 ** 31
        y, result = abs(x), 0

        while y != 0:
            result = result * 10 + y % 10
            if result > limit: return 0
            y //= 10

        return result if x >= 0 else -result

